python/python/glide/async_commands/core.py

```diff
@@ -1368,3 +1373,111 @@ async def invoke_script(
                 ["foo", "bar"]
         """
         return await self._execute_script(script.get_hash(), keys, args)
+
+    @dataclass
+    class PubSubMsg:
+        """Describes incoming message
+
+        Attributes:
+            message (str): Incoming message.
+            channel (str): Name of an channel that triggered the message.
+            pattern (Optional[str]): Pattern that triggered the message.
+            context (Optional[str]): User-provided context for this subscription.
```

Comment:
why is the context part of the message? IMO it's a break of encapsulation, and not a part of the message concept. The context should be passed as a second argument to the callback, not hidden inside the message.
Also, why is the context a string?

Reply:
I think it makes the usage more convenient - no need to access 2 params in the callback. Redis message concept does not mean it should be bit-exact to the API.
Optional[str] is a bug, will change to Optional[Any]
Done